File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cpp
class Solution {
public:
    int maximumLengthSubstring(string s) {
        int freq[26]={0};
        int l=0,ans=0;
        for(int r=0;r<s.length();r++){
            freq[s[r]-'a']++;
            while(freq[s[r]-'a']>2){
                freq[s[l]-'a']--;
                l++;
            }
            ans=max(ans,r-l+1);
        }
        return ans;
    }
};

File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        freq=defaultdict(int)
        l=0;ans=0
        for r in range(len(s)):
            freq[s[r]]+=1
            while freq[s[r]]>2:
                freq[s[l]]-=1
                l+=1
            ans=max(ans,r-l+1)
        return ans